File: Quiz_057.py
class converter:
    def __init__(self, decimal, base):
        self.decimal = decimal
        self.base = base

    def symbols(self, number):
        if number > 9:
            return chr(number + 55)

    def build(self):
        result = ""
        while self.decimal > 0:
            result = str(self.symbols(self.decimal % self.base)) + result
            self.decimal = self.decimal // self.base

        return result

test1 = converter(10, 16).build()
test2 = converter(10987, 20).build()
test3 = converter(10987, 36).build()

print(test1)
print(test2)
print(test3)

File: Quiz_057.js
class converter {
    constructor(decimal, base) {
        this.decimal = decimal
        this.base = base
    }

    symbols = (number) => {
        if (number > 9) {
            return String.fromCharCode(55 + number)
        }
        return number
    }

    convert = () => {
        let result = ""
        while (this.decimal > 0) {
            result = this.symbols(this.decimal % this.base) + result
            this.decimal = Math.floor(this.decimal / this.base)
        }
        return result
    }
}

let test1 = new converter(10, 16)
let test2 = new converter(10987, 20)
let test3 = new converter(10987, 36)

console.log(test1.convert())
console.log(test2.convert())
console.log(test3.convert())
